### PR TITLE
Add Rocket Pool Vault

### PR DESCRIPTION
**Pull request by chuddster**

### CODE REVIEW

File: constants/constants.ts [hunk 1]
```diff
@@ -172,6 +184,8 @@ export const WSTETH_PRICER = "0x4661951D252993AFa69b36bcc7Ba7da4a48813bF";
 
 export const SAVAX_PRICER = "0x0A59f35F00A482bB04d95428e1ec051cBac216C9";
 
+export const RETH_PRICER = "";
```
Comment (chuddster) [inline constants/constants.ts:187]: add chainlink reth pricer

File: contracts/interfaces/IRETHDepositPool.sol [hunk 2]
```diff
@@ -0,0 +1,6 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+interface IRETHDepositPool {
+    function deposit() external payable;
```
Comment (chuddster) [inline contracts/interfaces/IRETHDepositPool.sol:5]: https://etherscan.io/address/0x4D05E3d48a938db4b7a9A59A802D5b45011BDe58#code#F7#L99

File: contracts/interfaces/IRETH.sol [hunk 3]
```diff
@@ -0,0 +1,12 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IRETH is IERC20 {
```
Comment (chuddster) [inline contracts/interfaces/IRETH.sol:6]: https://etherscan.io/token/0xae78736cd615f374d3085123a210448e74fc6393

File: test/RibbonThetaVault.ts [hunk 4]
```diff
@@ -229,6 +232,39 @@ describe("RibbonThetaVault", () => {
...
+    amount: parseEther("20"),
+    contractOwnerAddress: RETH_OWNER_ADDRESS[chainId],
+  },
+  availableChains: [CHAINID.ETH_MAINNET],
+  protocol: OPTION_PROTOCOL.GAMMA,
+});
+
 type Option = {
   address: string;
   strikePrice: BigNumber;
```
Comment (chuddster) [inline test/RibbonThetaVault.ts:270]: add RETHDepositHelper.ts

File: scripts/deploy/21_reth_theta_vault.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  );
+
+  const initArgs = [
+    owner,
+    keeper,
+    feeRecipient,
+    MANAGEMENT_FEE,
+    PERFORMANCE_FEE,
+    "Ribbon rETH Theta Vault",
+    "rrETH-THETA",
```
Comment (fqlx) [inline scripts/deploy/21_reth_theta_vault.ts:57]: Should it be rETH or rrETH?

File: scripts/deploy/21_reth_theta_vault.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+    pricer.address,
+    strikeSelection.address,
+    PREMIUM_DISCOUNT,
+    AUCTION_DURATION,
+    {
+      isPut: false,
+      decimals: 18,
+      asset: WETH_ADDRESS[chainId],
```
Comment (fqlx) [inline scripts/deploy/21_reth_theta_vault.ts:65]: Should be RETH not WETH?

File: test/RibbonThetaVault.ts [hunk 7]
```diff
@@ -227,6 +230,39 @@ describe("RibbonThetaVault", () => {
     availableChains: [CHAINID.ETH_MAINNET],
     protocol: OPTION_PROTOCOL.TD,
   });
+
+  behavesLikeRibbonOptionsVault({
+    name: `Ribbon RETH Theta Vault (Call)`,
+    tokenName: "Ribbon RETH Theta Vault",
+    tokenSymbol: "rRETH-THETA",
+    asset: RETH_ADDRESS[chainId],
+    assetContractName: "IWETH",
```
Comment (fqlx) [inline test/RibbonThetaVault.ts:239]: Might be way the tests are failing, the contract abi is significantly different for RETH

File: scripts/deploy/21_reth_theta_vault.ts (hunk 5 above)
Comment (chuddster) [reply scripts/deploy/21_reth_theta_vault.ts:57]: rrETH

File: test/RibbonThetaVault.ts (hunk 7 above)
Comment (fqlx) [reply test/RibbonThetaVault.ts:239]: You already checked in the interface FYI just change it to `IRETH`
https://github.com/ribbon-finance/ribbon-v2/pull/255/files#diff-b75015944bb4beb6d69ef74161a50ad138c0f49378c122aef790b46940650106

Comment (chuddster) [reply test/RibbonThetaVault.ts:239]: should be fine - theta vault doesnt call any reth specific methods (only erc20 standard methods)

**Review by kenchangh — APPROVED** [review]

File: contracts/vaults/RETHVault/base/RibbonVault.sol [hunk 8]
```diff
@@ -0,0 +1,770 @@
...
+    }
+
+    /************************************************
+     *  DEPOSIT & WITHDRAWALS
+     ***********************************************/
+
+    /**
+     * @notice Deposits ETH into the contract and mint vault shares.
+     */
+    function depositETH() external payable nonReentrant {
```
Comment (chuddster) [inline contracts/vaults/RETHVault/base/RibbonVault.sol:304]: had to create new RibbonVault (vs using from base vault) since depositETH is not `virtual`

**Pull request merged**